### ISSUE TITLE
register_decoration place_on broken by registered ores

### ISSUE THREAD

**Issue by kakalak-lumberJack** [minetest/minetest#7609]
##### Issue type

 - Bug report
 
##### Minetest version

```
"0.4.17.1"
```

##### OS / Hardware

Operating system:
CPU: Linux

GPU model:
OpenGL version:

##### Summary

Decorations are placed on ore material though they are different from decoration's specified "place_on" field.
##### Steps to reproduce

Register an ore placed in default:dirt_with_grass. Flowers, grass, etc. will be placed on it.

**Comment by paramat** [comment]
Another order issue i think about is generating dungeons after generating ores, so that ores do not appear in dungeon walls (looks bad) when the dungeon uses local stone for walls.

**Comment by kakalak-lumberJack** [comment]
Yeah, I kind of figured it had to due with the order of decorations and ores being placed. I've also noticed the issue with dungeons. To me it seems logical that dungeons and ores should be first but I totally understand your caution. Thanks for looking at it.

**Comment by paramat** [comment]
Yes that order is ideal if possible. Ideally i would like:
Base terrain, biomes, caves, ores, dungeons, decorations, biome dust

**Comment by paramat** [comment]
^ That would also allow decorations to be placed in dungeons, so this is good if possible.

**Comment by HybridDog** [comment]
Why do you want biomes to be added before the caves?

**Comment by paramat** [comment]
Thinking on this, decorations should certainly come after ores, as some ore types are being used to place large solid volumes of nodes, not just a few scattered nodes. I can't think of any disadvantages either so i'll do this to get some testing time in before 5.0.0.

Checking, the current order is actually dungeons-decorations-ores, so decorations can already be placed in dungeons. I'll change this to ores-dungeons-decorations.

**Comment by paramat** [comment]
PR #7656
